### ISSUE TITLE
Date and time based filenames for the camera

### ISSUE THREAD

**Issue by davidrutland** [ubports/ubuntu-touch#1776]
### Description of the feature

Date and time based filenames for the camera eg: image20210414_141823091.jpg

 Right now, the count starts at IMG_00000001.jpg every time I clear out the storage.This is playing havoc with my UBsync > Nextcloud > Photoprism workflow as the names are unlikely to be unique.

 A date & time based filename would be far more useful (to me at least). If there's a way of doing this, I can't find it.

**Comment by Flohack74** [comment]
I like the idea, its also super-eas to implement. camera-app has moved here however: https://gitlab.com/ubports/apps/camera-app/-/issues you might want to create it there. That might speed up this improvement. Thanks!

**Comment by SevralT** [comment]
Cool idea. I like it.

**Comment by UniversalSuperBox** [comment]
That's interesting, photos from the camera on the Xperia X come out as `image20210421_161233207.jpg`. On which device are you having this trouble?

**Comment by Flohack74** [comment]
Maybe filename is created by Android HAL?

**Comment by fredldotme** [comment]
Either that or the Qt camera integration does it. I remember the names being different on devices where gst-droid was in use.

**Comment by mardy** [comment]
> Either that or the Qt camera integration does it. I remember the names being different on devices where gst-droid was in use.

Yes, but we could still do something about it in the camera app: if the name reported by the [imageSaved() signal](https://doc.qt.io/qt-5/qml-qtmultimedia-cameracapture.html#imageSaved-signal) does not contain the "AAAAMMDD" string, it could rename it to a pattern that contains it.
